src/test/java/com/at/globalclasses/MongoDBConnection.java

```diff
@@ -87,12 +87,16 @@ public boolean compareID(String collection, String id) {
 
     public String getObjectByID(String collection, String id) {
         String mObject = "";
+        JSONObject json = new JSONObject(id);
+        String idValue = json.getString("id");
         MongoCollection<Document> coll = mDataBase.getCollection(collection);
-        FindIterable<Document> findIterable = coll.find(Filters.eq("_id", new ObjectId(id)));
+        ObjectId objectId = new ObjectId(idValue);
+        FindIterable<Document> findIterable = coll.find(Filters.eq("_id", objectId));
```

Comment:
You could have left the method just as it was. These lines that you added are not needed.

Reply:
I put it the way it was before in the new commit